File: src/test/java/no/fintlabs/mapping/FormattingUtilsServiceTest.java
package no.fintlabs.mapping;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.junit.jupiter.MockitoExtension;

import static org.assertj.core.api.AssertionsForClassTypes.assertThat;

@ExtendWith(MockitoExtension.class)
class FormattingUtilsServiceTest {

    FormattingUtilsService formattingUtilsService;

    @BeforeEach
    public void setUp() {
        formattingUtilsService = new FormattingUtilsService();
    }

    @Test
    public void givenNullEmailShouldReturnInvalidEmailString() {
        String formattedEmail = formattingUtilsService.extractEmailDomain(null);

        assertThat(formattedEmail).isEqualTo("Invalid email");
    }

    @Test
    public void givenInvalidEmailShouldReturnInvalidEmailString() {
        String email = "test";
        String formattedEmail = formattingUtilsService.extractEmailDomain(email);

        assertThat(formattedEmail).isEqualTo("Invalid email");
    }

    @Test
    public void givenValidEmailShouldReturnEmailDomain() {
        String email = "test@test.no";
        String formattedEmail = formattingUtilsService.extractEmailDomain(email);

        assertThat(formattedEmail).isEqualTo("test.no");
    }

    @Test
    public void givenNullStringShouldReturnNull() {
        String formattedKommunenavn = formattingUtilsService.formatKommunenavn(null);

        assertThat(formattedKommunenavn).isEqualTo(null);
    }

    @Test
    public void givenEmptyStringShouldReturnEmptyString() {
        String kommunenavn = "";
        String formattedKommunenavn = formattingUtilsService.formatKommunenavn(kommunenavn);

        assertThat(formattedKommunenavn).isEqualTo("");
    }

    @Test
    public void givenUpperCaseStringShouldReturnFormattedStringWithCapitalizedFirstLetter() {
        String kommunenavn = "SKIEN";
        String formattedKommunenavn = formattingUtilsService.formatKommunenavn(kommunenavn);

        assertThat(formattedKommunenavn).isEqualTo("Skien");
    }

    @Test
    public void givenUpperCaseStringWithOneDashBetweenWordsShouldReturnCapitalizedFirstLetterOnBothWords() {
        String kommunenavn = "NORD-ODAL";
        String formattedKommunenavn = formattingUtilsService.formatKommunenavn(kommunenavn);

        assertThat(formattedKommunenavn).isEqualTo("Nord-Odal");
    }

}

File: src/test/kotlin/no/novari/flyt/egrunnerverv/gateway/instance/mapping/FormattingUtilsServiceTest.kt
package no.novari.flyt.egrunnerverv.gateway.instance.mapping

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class FormattingUtilsServiceTest {
    private lateinit var formattingUtilsService: FormattingUtilsService

    @BeforeEach
    fun setUp() {
        formattingUtilsService = FormattingUtilsService()
    }

    @Test
    fun `extractEmailDomain returns null for invalid email`() {
        assertThat(formattingUtilsService.extractEmailDomain("test")).isNull()
        assertThat(formattingUtilsService.extractEmailDomain("")).isNull()
    }

    @Test
    fun `extractEmailDomain returns email domain for valid email`() {
        assertThat(formattingUtilsService.extractEmailDomain("test@test.no")).isEqualTo("test.no")
    }

    @Test
    fun `formatKommunenavn returns null for null input`() {
        assertThat(formattingUtilsService.formatKommunenavn(null)).isNull()
    }

    @Test
    fun `formatKommunenavn preserves blank input`() {
        assertThat(formattingUtilsService.formatKommunenavn("")).isEqualTo("")
    }

    @Test
    fun `formatKommunenavn capitalizes and normalizes whitespace`() {
        assertThat(
            formattingUtilsService.formatKommunenavn("NORD-ODAL\tDONALD\nDUCK"),
        ).isEqualTo("Nord-Odal Donald Duck")
    }
}
